bug fixes

Berthalamew opened the pull request:
- fix drop logic
- fix bug where the console wouldn't display any output
- fix title on multiple game instances

Berthalamew committed:
misc changes

- fix drop logic
- convert some macro constants to an enum
-  make some map_repository functions static

Berthalamew committed:
fix bug where the console wouldn't display any output

- disable formatting for console commands
- change exception output so we print the string instead of formatting it again

Berthalamew committed:
fix title on multiple game instances

- fix port number incrementation for config
- fix args on ustrncat

nukeulater commented on xlive/H2MOD/GUI/imgui_integration/Console/CommandCollection.cpp:
```diff
@@ -225,15 +225,15 @@ static int CommandCollection::BoolVarHandlerCmd(const std::vector<std::string>&
 	{
 		booleanCmdVar->SetVal(!booleanCmdVar->GetVal());
-		outputCb(StringFlag_None, "%s", booleanCmdVar->AsString().c_str());
+		outputCb(StringFlag_None, booleanCmdVar->AsString().c_str());
```
why wouldn't this work anymore??

Berthalamew replied:
It works fine either way.
Wouldn't we want to pass the format as the string itself?
Rather than passing a specifier for just a string and then the string?

nukeulater replied:
below there are instances where the text has some spacing, so not sure why you would change this.

Berthalamew replied:
Thought it was a mistake as it was using tab within the source code vs the tab character.
I'll change it to use the tab character.

Berthalamew committed:
requested changes

- restore indentation for exceptions (changed to tab character vs tab in the string itself)

nukeulater commented:
this text is not getting logged anymore:  [Reading H2Configuration file...](https://github.com/pnill/cartographer/pull/832/files#diff-76733ff73f98ccaf2c189ba693986c55a6509c1613f62d1020902ecd182056d0R569)

Berthalamew committed:
fix onscreendebug logging in config

Berthalamew commented:
fixed

Berthalamew committed:
initialize Command collection commands on dedis

Berthalamew committed:
fix dedis being unjoinable on lan

- check "-instance" instead of "instance"
- add dedi offset for timing_initialize

Berthalamew committed:
better formatting for unknown command message
